src/libostree/ostree-repo-static-delta-core.c

```diff
@@ -409,7 +409,7 @@ ostree_repo_static_delta_execute_offline_with_signature (OstreeRepo *self, GFile
         return glnx_throw_errno_prefix (error, "openat(O_DIRECTORY)");
       else
         {
-          g_autofree char *dir = dirname (g_strdup (dir_or_file_path));
+          g_autofree char *dir = g_path_get_dirname (g_strdup (dir_or_file_path));
```

Comment:
In this case, you don't need to duplicate `dir_or_file_path`. `g_path_dirname` takes care of it. This is essentially copying the buffer twice with the first copy being leaked.